src/lib/localStore.ts

```diff
@@ -0,0 +1,82 @@
+import * as extension from 'extensionizer';
+import log from 'loglevel';
+import { REVERSE_MIGRATIONS } from 'lib/reverseMigrations';
+
+const CURRENT_MIGRATION_VERSION = 0;
+
+export default class ExtensionStore {
+  public isSupported: boolean;
+
+  constructor() {
+    this.isSupported = !!extension.storage.local;
+
+    if (!this.isSupported) {
+      log.error('Storage local API not available.');
+    }
+  }
+
+  async get() {
+    if (!this.isSupported) return undefined;
+    const result = await this._get();
+
+    if (isEmpty(result)) {
+      return undefined;
+    } else {
+      return result;
+    }
+  }
+
+  async set(state: Record<string, unknown>) {
+    return this._set(state);
+  }
+
+  async migrate() {
+    const { migrationVersion } = await this._get();
+    const version = (migrationVersion as number) || 0;
+
+    await this._set({
+      migrationVersion: CURRENT_MIGRATION_VERSION,
+    });
+
+    if (version > CURRENT_MIGRATION_VERSION) {
```

Comment:
I think we need some fallback mechanism in case one of the `reverseMigrate` fails.